tools/ontology-builder/src/compute_descendent_mappings.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python
+"""
+# Descendant Mappings for Tissues and Cell Types
+
+## Overview
+
+The ontology-aware tissue and cell type filters in the Single Cell Data Portal each require artifacts generated
+by this script.
+
+#### Descendant Mappings
+To facilitate in-filter, cross-panel restriction of filter values, a descendant hierarchy dictionary is required by
+the Single Cell Data Portal frontend. For example, if a user selects `hematopoietic system` in the tissue filter's
+`System` panel, the values in the tissue filter's `Organ` and `Tissue` panels must be restricted by `hematopoietic
+system`.
+
+This script generates a dictionary of descendants keyed by tissue or cell type ontology term ID. The dictionary
+is stored as a JSON file and copied to cellxgene-ontology-guide/ontology_assets directory. A versioned github release is
+ created to simplify referencing in the Single Cell Data Portal.
+
+The descendant mappings should be updated when:
+
+1. The ontology version is updated,
+2. A new tissue or cell type is added to the production corpus, or,
+3. The hand-curated systems, organs, cell classes or cell subclasses are updated.
+"""
+
+import json
+import os
+from typing import Any, Dict, List
+from urllib.request import urlopen
+
+import env
+from cellxgene_ontology_guide.ontology_parser import OntologyParser
+
+
+def load_prod_datasets() -> Any:
+    """
+    Request datasets the production corpus.
+    """
+    response = urlopen("https://api.cellxgene.cziscience.com/dp/v1/datasets/index").read().decode("utf-8")
+    return json.loads(response)
+
+
+def extract_cell_types(datasets: List[Dict[str, Any]]) -> List[str]:
+    """
+    List the set of cell type values for the given datasets.
+
+    :param datasets: a list of datasets from the production corpus.
+    :return: a list formated of cell type values
+    """
+    cell_types = set()
+    for dataset in datasets:
+        for cell_type in dataset["cell_type"]:
+            cell_types.add(cell_type["ontology_term_id"].replace("_", ":", False))
```

Comment:
Don't think we need that 'False' arg for str replace, the third arg type is an optional int 

https://docs.python.org/3/library/stdtypes.html#str.replace